sigridci/sigridci/azure_pull_request_report.py

```diff
@@ -47,25 +55,46 @@ def isSupported(self, options):
             "SYSTEM_PULLREQUEST_PULLREQUESTID" in os.environ and \
             options.runMode == RunMode.FEEDBACK_ONLY
 
-    def buildURL(self):
+    def findExistingSigridCommentThreadId(self):
+        existingThreads = json.load(self.callAzure("GET", None, None))
+
+        for thread in existingThreads["value"]:
+            for comment in thread["comments"]:
+                if comment["content"].startswith("# Sigrid"):
+                    return thread["id"]
+
+        return None
+
+    def callAzure(self, method, body, threadId):
+        request = urllib.request.Request(self.buildURL(threadId), json.dumps(body).encode("utf-8"))
+        request.method = method
+        request.add_header("Authorization", f"Bearer {os.environ['SYSTEM_ACCESSTOKEN']}")
+        request.add_header("Content-Type", "application/json")
+        return urllib.request.urlopen(request)
+
+    def buildURL(self, threadId):
         baseURL = os.environ["SYSTEM_TEAMFOUNDATIONCOLLECTIONURI"]
         project = os.environ["SYSTEM_TEAMPROJECTID"]
         repo = os.environ["BUILD_REPOSITORY_NAME"]
         pr = os.environ["SYSTEM_PULLREQUEST_PULLREQUESTID"]
         version = self.AZURE_API_VERSION
-        return f"{baseURL}{project}/_apis/git/repositories/{repo}/pullRequests/{pr}/threads?api-version={version}"
 
-    def buildRequestBody(self, feedbackFile):
+        if threadId:
+            return f"{baseURL}{project}/_apis/git/repositories/{repo}/pullRequests/{pr}/threads/{threadId}?api-version={version}"
```

Comment:
What's the reason we're using different versions? I think that newer versions typically _extend_ existing endpoints, so we could as well choose the lowest version that has this endpoint (incl. the query parameters we need).

Reply:
Made it more clear by directly using the constant.